### ISSUE TITLE
Permanent UI test failure - org.mozilla.samples.glean.MainActivityTest - checkGleanClickData

### ISSUE THREAD

**Issue by Archaeopteryx** [mozilla-mobile/android-components#10651]
Ui tests were broken for several issues due to another issue. Could this permanent failure stem from #10335 @pocmo ?

### Firebase Test Run:
https://console.firebase.google.com/project/moz-android-components-230120/testlab/histories/bh.acacddb44ee28c64/matrices/8796090481347570318/executions/bs.d128e3ce5b5e33d3/testcases/2
### Stacktrace:
```
java.lang.AssertionError: To use the testing API, apply the GleanTestRule to set up a disposable Glean instance. e.g. GleanTestRule(ApplicationProvider.getApplicationContext())
at mozilla.telemetry.glean.Dispatchers$WaitableCoroutineScope.assertInTestingMode(Dispatchers.kt:76)
at mozilla.telemetry.glean.private.CounterMetricType.testHasValue(CounterMetricType.kt:102)
at mozilla.telemetry.glean.private.CounterMetricType.testHasValue$default(CounterMetricType.kt:100)
at org.mozilla.samples.glean.MainActivityTest.checkGleanClickData(MainActivityTest.kt:30)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:154)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:395)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
```
### Build: 
https://github.com/mozilla-mobile/android-components/commit/1772620e8b06c12061cef67fd7c7ae0e8b9fc65d

**Comment by pocmo** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1721675

**Comment by pocmo** [comment]
Fixed by 465b85378d32d9f921bab2b7650edcfd3cecc8f7
